External/Notepad/Storage.swift

```diff
@@ -191,6 +199,12 @@ class Storage: NSTextStorage {
     func refreshStyle(markdownEnabled: Bool) {
         self.theme = Theme(markdownEnabled: markdownEnabled)
     }
+
+    func endEditingWithoutRestyling() {
+        skipRestyling = true
+        endEditing()
+        skipRestyling = false
```

Comment:
Kudos to you, sir. This approach is much simpler than what I initially had in mind!!